### PR TITLE
Fixed handicap glitch in infection

### PR DESCRIPTION
**Pull request by nukeulater**
OG gametype now runs on dedicated servers too, setting them to 30hz, also don't run 60 tick patches when this mod is on.
Run sunflare fix only on startup, because it's a constant which never gets modified by the game code.
Some misc cleanup.
Added fix by himanshu01.

### CODE REVIEW

**Comment by iSupersniper** [comment]
Please update this merge request and create a dll to test in dev preview.

File: xlive/H2MOD/Variants/VariantPlayer.cpp [hunk 1]
```diff
@@ -57,7 +57,7 @@ DatumIndex VariantPlayer::GetUnitDatum(XUID xuid)
 
 	if (xuid_to_unit_datum.find(xuid) == xuid_to_unit_datum.end())
 	{
-		for (auto i = 0; i < 15; i++)
+		for (auto i = 0; i < 16; i++)
```
Comment (shayaantx) [inline xlive/H2MOD/Variants/VariantPlayer.cpp:60]: why is this 16 now?

if this is valid, add a constant for this value, explaining why its "16"

File: xlive/H2MOD/Modules/Networking/Networking.cpp [hunk 2]
```diff
@@ -459,14 +478,23 @@ void applyConnectionPatches()
...
+		// for whatever reason this code will not execute, marking GetAddress not inline or exec in debug mode fixes it
+		//WritePointer((DWORD)h2mod->GetAddress(addresses[i], addresses_dedi[i]) + 4, &unk_flt_); 
 	}
 
 	// increase max bits per second of LIVE netcode (3000 bytes -> ~8000 bytes)
-	WriteValue<DWORD>((DWORD)h2mod->GetAddress(0x1AAD63 + 6, 0x1AB268 + 6), 61440);
+	WriteValue<DWORD>((DWORD)h2mod->GetAddress(0x1AAD63, 0x1AB268) + 6, 61440);
+
+	BYTE call[] = { 0xE8 };
+	DWORD network_observer_patch_addr = (DWORD)h2mod->GetAddress(0x1BF1B9, 0x1B9093);
```
Comment (shayaantx) [inline xlive/H2MOD/Modules/Networking/Networking.cpp]: please add a small doc here or on the method explaining why/what this does

File: xlive/H2MOD.cpp [hunk 3]
```diff
@@ -1089,83 +1083,90 @@ void __cdecl OnMapLoad(int a1)
...
-		H2Tweaks::sunFlareFix();
-
 		//H2Tweaks::applyShaderTweaks(); 
 
 		if (GameState == 3)
 		{
 			// send server map checksums to client
 			//MapChecksumSync::SendState();
 			//inform players of the current advanced lobby settings
+			// TODO, now we got good custom packets ;)
```
Comment (shayaantx) [inline xlive/H2MOD.cpp]: put an issue in to reenable this and redo your TODO-ISSUE_NUMBER

File: xlive/H2MOD.cpp [hunk 4]
```diff
@@ -10,14 +10,12 @@
...
 #include "H2MOD\Modules\Config\Config.h"
-#include "H2MOD\Modules\Startup\Startup.h"
 #include "H2MOD\Modules\MainMenu\Ranks.h"
 #include "H2MOD\Variants\H2X\H2X.h"
 #include "H2MOD\Variants\GunGame\GunGame.h"
-#include "XLive\UserManagement\CUser.h"
 #include "H2MOD\Modules\Networking\Memory\bitstream.h"
 #include "H2MOD\Modules\Networking\CustomPackets\CustomPackets.h"
-#include "H2MOD\Modules\Networking\NetworkSession\NetworkSession.h"
+#include <Shlwapi.h> /*StrStrIW function in OnMapLoad*/
```
Comment (shayaantx) [inline xlive/H2MOD.cpp]: this doc seems unnecessary

File: xlive/H2MOD/Variants/VariantPlayer.cpp (hunk 1 above)
Comment (nukeulater) [reply xlive/H2MOD/Variants/VariantPlayer.cpp:60]: commit description clarifies

File: xlive/H2MOD.cpp (hunk 4 above)
Comment (nukeulater) [reply xlive/H2MOD.cpp]: i didn't add that, will change

File: xlive/H2MOD/Modules/Networking/Networking.cpp (hunk 2 above)
Comment (nukeulater) [reply xlive/H2MOD/Modules/Networking/Networking.cpp]: places call, will fix

File: xlive/H2MOD/Variants/VariantPlayer.cpp (hunk 1 above)
Comment (shayaantx) [reply xlive/H2MOD/Variants/VariantPlayer.cpp:60]: The commit description is "Fixed player loop only going up to index 14 inside VariantPlayer inst……ead of 15 (player 16)-Himanshu01"

That doesn't really explain anything, don't rely on commit descriptions to describe your code. Please make special fields like this a constant with a simple doc.

**Pull request merged**